### ISSUE TITLE
Advanced trading sell lists item types wrong

### ISSUE THREAD

**Issue by ndavalos** [EpochModTeam/DayZ-Epoch#1844]
So if you have weapons and magazines that have the same class name advanced trading will default to considering the item a magazine when creating the sell lists.

You can see this with Javelins and Stingers most notably. The ammo for a Javelin is also the same class name as the launcher. Unfortunately the way advanced trading sell list building is structured in z_at_checkArrayInConfig.sqf, it will look in the trader categories to find out what the type of item is (magazine, weapon etc). It really should look at where it came from, ie. getMagazineCargo, getWeaponCargo, etc.

So basically if you have a weapon that has the same class name as it's ammo you end up always having the magazine class added to the sell list, not the weapon class. When you try and sell it, it fails.
